pkg/signature/kms/azure/integration_test.go

```diff
@@ -0,0 +1,184 @@
+//go:build integration
+
+//
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package azure contains utilities related to Microsoft Azure KMS.
+package azure
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/google/go-cmp/cmp"
+)
+
+/*
+The following environment variables must be set:
+AZURE_KEY_REF - full azure key reference in the format azurekms://[Key Vault Name].vault.azure.net/[Key Name](/[Key Version]Optional)
+KEY_NAME - Azure key name
+VAULT_URL - Azure Vault URL
+*/
+
+func TestMain(m *testing.M) {
+	azureKeyRef := os.Getenv("AZURE_KEY_REF")
+	if azureKeyRef == "" {
+		panic("AZURE_KEY_REF must be set")
+	}
+	os.Exit(m.Run())
+}
+
+func TestGetAzClientOpts(t *testing.T) {
+	testCases := []struct {
+		env            string
+		expectedConfig cloud.Configuration
+	}{{
+		env:            "AZUREUSGOVERNMENT",
+		expectedConfig: cloud.AzureGovernment,
+	}, {
+		env:            "AZUREUSGOVERNMENTCLOUD",
+		expectedConfig: cloud.AzureGovernment,
+	}, {
+		env:            "AZURECHINACLOUD",
+		expectedConfig: cloud.AzureChina,
+	}, {
+		env:            "AZURECLOUD",
+		expectedConfig: cloud.AzurePublic,
+	}, {
+		env:            "AZUREPUBLICCLOUD",
+		expectedConfig: cloud.AzurePublic,
+	}, {
+		env:            "",
+		expectedConfig: cloud.AzurePublic,
+	}}
+
+	for _, tc := range testCases {
+		err := os.Setenv("AZURE_ENVIRONMENT", tc.env)
```

Comment:
does this work with `t.Setenv` instead of `os.Setenv`? using `t.Setenv` will ensure it gets unset after the test case is run.

Reply:
Yes, just tested that this works with `t.Setenv`. I'll push an update.